e2e/fbproxy/app/middleware.go

```diff
@@ -0,0 +1,78 @@
+package app
+
+import (
+	"context"
+	"encoding/json"
+	"math/big"
+
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/log"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+type txSigner interface {
+	Sign(ctx context.Context, digest common.Hash, signer common.Address) ([65]byte, error)
+}
+
+// NewSendTxMiddleware returns a middleware func that
+//   - intercepts eth_sendTransaction requests, signs them `txsigner`
+//     and replaces them with eth_sendRawTransaction requests
+//   - leaves all other requests unmodified
+func NewSendTxMiddleware(txsigner txSigner, chainID *big.Int) Middleware {
```

Comment:
Exported for tests - should we just keep private and use `_internal_test` pattern?